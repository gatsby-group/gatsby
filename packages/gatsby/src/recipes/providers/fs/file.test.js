const file = require(`./file`)

const root = __dirname
const content = `Hello, world!`

test(`create writes a file`, async () => {
  const filePath = `bar.txt`

  await file.create({ root }, { path: filePath, content })

  const result = await file.read({ root }, { path: filePath })

  expect(result.content).toEqual(content)

  await file.destroy({ root }, { path: filePath })
})

test(`update overwrites a file`, async () => {
  const filePath = `bar.txt`
  const newContent = `new content!`

  await file.create({ root }, { path: filePath, content })
  await file.update({ root }, { path: filePath, content: newContent })

  const result = await file.read({ root }, { path: filePath })

  expect(result.content).toEqual(newContent)

  await file.destroy({ root }, { path: filePath })
})
