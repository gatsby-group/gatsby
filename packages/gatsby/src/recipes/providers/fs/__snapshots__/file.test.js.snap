// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file resource e2e file resource test: File create 1`] = `
Object {
  "_message": "Wrote file file.txt",
  "content": "Hello, world!",
  "id": "file.txt",
  "path": "file.txt",
}
`;

exports[`file resource e2e file resource test: File create plan 1`] = `
Object {
  "currentState": "",
  "describe": "Write file.txt",
  "diff": "[36m@@ -1 +1 @@[m
[31m-[m
[32m+[m[32mHello, world![m
",
  "newState": "Hello, world!",
}
`;

exports[`file resource e2e file resource test: File destroy 1`] = `
Object {
  "_message": "Wrote file file.txt",
  "content": "Hello, world!1",
  "id": "file.txt",
  "path": "file.txt",
}
`;

exports[`file resource e2e file resource test: File update 1`] = `
Object {
  "_message": "Wrote file file.txt",
  "content": "Hello, world!1",
  "id": "file.txt",
  "path": "file.txt",
}
`;

exports[`file resource e2e file resource test: File update plan 1`] = `
Object {
  "currentState": "Hello, world!",
  "describe": "Write file.txt",
  "diff": "[36m@@ -1 +1 @@[m
[31m-Hello, world![m
[32m+[m[32mHello, world!1[m
",
  "newState": "Hello, world!1",
}
`;
